OpenFn/Lightning
# Enable adaptor icons to load gracefully

## Notes for the reviewer

This PR fixes the load of adaptor icons in the Workflow Diagram so that data is shared between all components, and the icons load "gracefuly".

The fix is to use a hook which shares a single promise between all consuming components. The first call to the hook creates a promise to fetch and parse the icon manifest. Subsequent calls will just add a `then()` to this promise chain. After the data has been downloaded once, the promise will resolve instantly.

If the icon manifest changes during a user session, the icons will not update. A refresh of the page will download the latest icons, though.

Note that in the event of an error (ie the icons failing to load), the component will still fallback nicely to textural representation. An error will be logged to the console.

## QA Notes

If your internet connection is fast enough you won't see the issue - but if you look at the Diagram on `main` and check the network log, you'll notice that `adaptor_icons.json` is downloaded once per job node. On slow connections or large charts, this causes icons to pop in one at a time.

Against this branch, you should only see `adaptor_icons.json` be downloaded once.

## Related issue

Fixes #1140

## Review checklist
